### PR TITLE
Add `tablist.always_visible` setting (#497)

### PR DESCRIPTION
**Pull request by hofheinz**

### CODE REVIEW

File: lib/lousy/widget/tablist.lua [hunk 1]
```diff
@@ -211,6 +216,13 @@ function _M.new(notebook, orientation)
     return tlist
 end
 
+settings.register_settings({
+    ["tablist.always_visible"] = {
+        type = "boolean",
+        default = false,
```
Comment (aidanholm) [inline lib/lousy/widget/tablist.lua:221]: This also needs `domain_specific = false,`

File: lib/lousy/widget/tablist.lua [hunk 2]
```diff
@@ -123,8 +124,12 @@ function _M.new(notebook, orientation)
...
     -- Attach notebook signal handlers
     local function update_tablist_visibility()
-        if tlist.visible and notebook:count() >= 2 then tlist.widget:show() end
-        if not tlist.visible or notebook:count() < 2 then tlist.widget:hide() end
+        if tlist.visible and (notebook:count() >= 2 or settings.tablist.always_visible) then
+            tlist.widget:show()
+        end
+        if not tlist.visible or (notebook:count() < 2 and not settings.tablist.always_visible) then
+            tlist.widget:hide()
+        end
```
Comment (aidanholm) [inline lib/lousy/widget/tablist.lua]: I'd prefer that this be changed to set `tlist.widget.visible = true/false` instead; it's much clearer:
```lua
if settings.get_setting("tablist.always_visible") then
    tlist.widget.visible = true
else
    tlist.widget.visible = notebook:count() > 1
end
```

Comment (aidanholm) [reply lib/lousy/widget/tablist.lua]: Note that to be really robust, we need to call `update_tablist_visibility` when the `"tablist.always_visible"` setting changes, but I haven't added a `setting-changed` signal yet.

File: lib/lousy/widget/tablist.lua [hunk 3]
```diff
@@ -123,8 +124,12 @@ function _M.new(notebook, orientation)
 
     -- Attach notebook signal handlers
     local function update_tablist_visibility()
-        if tlist.visible and notebook:count() >= 2 then tlist.widget:show() end
-        if not tlist.visible or notebook:count() < 2 then tlist.widget:hide() end
+        if tlist.visible and (notebook:count() >= 2 or settings.tablist.always_visible) then
```
Comment (aidanholm) [inline lib/lousy/widget/tablist.lua]: Don't use `settings.tablist.always_visible`; this is an API for the user, not for plugins. It won't return the default value, for example, and it's slower (not that that should matter much).

**Review by aidanholm — CHANGES_REQUESTED** [review]
Thanks for the PR; looks good, but some minor changes are needed

**Comment by hofheinz** [comment]
Thank you for the code review and the constructive comments! I have updated the commit (and taken your version of the `update_tablist_visibility` method, works as expected). I agree with your remark about robustness (what if `tablist.always_visible` gets updated while one tab is open?). I could add a "todo" comment at the `update_tablist_visibility` method, and/or I can try to keep an eye on future commits (to see when the right signal has been added and change the method then).

Please allow one question about your version of `update_tablist_visibility`: you always set the `tlist.visible` property (to true or false), even if it already had that same value. In contrast, the previous code (not mine of course) went out of its way to only set `tlist.visible` if that changed its value (from true to false, or from false to true). Was there a reason for that? (Like: even changing from true to true triggers all kinds of then-unnecessary signals or such.)

**Comment by aidanholm** [comment]
I'll probably add the code to listen for setting changes when that's finished; don't worry about it.

The way the old code worked was mostly just because of the way using method calls encourages a particular code style. It shouldn't have any performance impact, and if it does I'd consider that a bug in the widget implementation (which would be easy enough to fix).

**Comment by hofheinz** [comment]
Thanks for the explanation!

**Comment by aidanholm** [comment]
No worries, questions always welcome :)

**Comment by aidanholm** [comment]
Thanks for the PR!

**Pull request merged**